src/App.tsx

```diff
@@ -1,62 +1,135 @@
+    setSelectedUser(user);
+    setIsLoading(true);
+    setSelectedPost(undefined);
+  };
+
+  const handleSelectedPost = (post: Post) => {
+    setSelectedPost(post);
+  };
+
+  const handleClosedPost = () => {
+    setSelectedPost(undefined);
+  };
+
   return (
     <main className="section">
       <div className="container">
         <div className="tile is-ancestor">
           <div className="tile is-parent">
             <div className="tile is-child box is-success">
               <div className="block">
-                <UserSelector />
+                <UserSelector
+                  users={users}
+                  onUserClick={handleSelectedUser}
+                />
               </div>
 
               <div className="block" data-cy="MainContent">
-                <p data-cy="NoSelectedUser">
-                  No user selected
-                </p>
+                {!selectedUser && (
+                  <p data-cy="NoSelectedUser">
+                    No user selected
+                  </p>
+                )}
 
-                <Loader />
+                {isLoading && (
+                  <Loader />
+                )}
 
-                <div
-                  className="notification is-danger"
-                  data-cy="PostsLoadingError"
-                >
-                  Something went wrong!
-                </div>
+                {!isLoading && loadingError && (
 
-                <div className="notification is-warning" data-cy="NoPostsYet">
-                  No posts yet
-                </div>
+                  <NotificationMessage
+                    notification={error}
+                    errorType="is-danger"
+                    data="PostsLoadingError"
+                  />
+                )}
 
-                <PostsList />
+                {!isLoading && !loadingError
+                && selectedUser && noPosts && (
+                  <NotificationMessage
+                    notification={error}
+                    errorType="is-warning"
+                    data="NoPostsYet"
+                  />
+                )}
+
+                {!isLoading && !loadingError
+                && selectedUser && !noPosts && (
+                  <PostsList
+                    posts={userPosts}
+                    onPostOpen={handleSelectedPost}
+                    onPostClose={handleClosedPost}
+                  />
+                )}
               </div>
             </div>
           </div>
 
-          <div
-            data-cy="Sidebar"
-            className={classNames(
-              'tile',
-              'is-parent',
-              'is-8-desktop',
-              'Sidebar',
-              'Sidebar--open',
-            )}
-          >
-            <div className="tile is-child box is-success ">
-              <PostDetails />
-            </div>
-          </div>
+          {selectedPost && (
+            <Sidebar
+              post={selectedPost}
+            />
```

Comment:
```suggestion
            <Sidebar post={selectedPost} />
```
It is better to make it in one line if there are < 3 props and the line is not too long